drivers/i2c/i2c_mspm0g3xxx.c

```diff
@@ -736,6 +740,33 @@ static int i2c_mspm0g3xxx_init(const struct device *dev)
 	struct i2c_mspm0g3xxx_data *data = dev->data;
 	int ret;
 
+
+	/** from dl_i2c.h
+	 *  scl_period = (1 + tpr) * (scl_lp + scl_hp) * int_clk_prd
+	 *
+	 *  where:
+	 *  scl_prd is the scl line period (i2c clock)
+	 *
+	 *  tpr is the timer period register value (range of 1 to 127)
+	 *
+	 *  scl_lp is the scl low period (fixed at 6)
+	 *  scl_hp is the scl high period (fixed at 4)
+	 *
+	 *  clk_prd is the functional clock period in ns
+	 *
+	 *  what we are setting is tpr. so is we solve the equation we end
+	 *  up with : tpr = (int_clk_rate / (scl_lp + scl_hp) * scl_rate) - 1
+	 */
```

Comment:
This comment is kinda floating on its own now. I think it's adequate to include the explanation in the commit message rather than trying to stick this comment into a macro.

Alternatively you could define the formula as a macro itself, include the comment right above it, and then use the macro within the config-creation stuff at the bottom.

Reply:
Putting the comment in a macro kinda defeats the purpose of comments IMO. you want them to be local to where the actual code is. I can move the comment to the commit message.